src/colmap/estimators/bundle_adjustment.cc

```diff
@@ -432,141 +377,33 @@ ceres::Solver::Options BundleAdjuster::SetUpSolverOptions(
-          cam_from_world_rotation,
-          cam_from_world_translation,
-          point3D.xyz.data(),
-          camera_params);
-    }
-  }
-
-  if (num_observations > 0) {
-    camera_ids_.insert(image.CameraId());
-
-    // Set pose parameterization.
-    if (!constant_cam_pose) {
-      SetQuaternionManifold(problem_.get(), cam_from_world_rotation);
-      if (config_.HasConstantCamPositions(image_id)) {
-        const std::vector<int>& constant_position_idxs =
-            config_.ConstantCamPositions(image_id);
-        SetSubsetManifold(3,
-                          constant_position_idxs,
-                          problem_.get(),
-                          cam_from_world_translation);
-      }
-    }
-  }
-}
-
-void BundleAdjuster::AddPointToProblem(const point3D_t point3D_id,
-                                       Reconstruction* reconstruction,
-                                       ceres::LossFunction* loss_function) {
-  Point3D& point3D = reconstruction->Point3D(point3D_id);
-
-  // Is 3D point already fully contained in the problem? I.e. its entire track
-  // is contained in `variable_image_ids`, `constant_image_ids`,
-  // `constant_x_image_ids`.
-  if (point3D_num_observations_[point3D_id] == point3D.track.Length()) {
-    return;
-  }
-
-  for (const auto& track_el : point3D.track.Elements()) {
-    // Skip observations that were already added in `FillImages`.
-    if (config_.HasImage(track_el.image_id)) {
-      continue;
-    }
-
-    point3D_num_observations_[point3D_id] += 1;
-
-    Image& image = reconstruction->Image(track_el.image_id);
-    Camera& camera = *image.CameraPtr();
-    const Point2D& point2D = image.Point2D(track_el.point2D_idx);
-
-    // CostFunction assumes unit quaternions.
-    image.CamFromWorld().rotation.normalize();
-
-    // We do not want to refine the camera of images that are not
-    // part of `constant_image_ids_`, `constant_image_ids_`,
-    // `constant_x_image_ids_`.
-    if (camera_ids_.count(image.CameraId()) == 0) {
-      camera_ids_.insert(image.CameraId());
-      config_.SetConstantCamIntrinsics(image.CameraId());
-    }
-    problem_->AddResidualBlock(
-        CreateCameraCostFunction<ReprojErrorConstantPoseCostFunctor>(
-            camera.model_id, point2D.xy, image.CamFromWorld()),
-        loss_function,
-        point3D.xyz.data(),
-        camera.params.data());
-  }
-}
-
-void BundleAdjuster::ParameterizeCameras(Reconstruction* reconstruction) {
-  const bool constant_camera = !options_.refine_focal_length &&
-                               !options_.refine_principal_point &&
-                               !options_.refine_extra_params;
-  for (const camera_t camera_id : camera_ids_) {
-    Camera& camera = reconstruction->Camera(camera_id);
-
-    if (constant_camera || config_.HasConstantCamIntrinsics(camera_id)) {
-      problem_->SetParameterBlockConstant(camera.params.data());
-      continue;
+void ParameterizeCameras(const BundleAdjustmentOptions& options,
```

Comment:
This function is specific for bundle adjustment. If we make it no longer a class method, should we add another level of namespace to avoid potential collapsing in the future?

Reply:
Same applies to other functions in this file.

Reply:
I probably don't fully understand but it is already in an anonymous namespace. Is this what you wanted to suggest?

Reply:
oh i missed that. Thanks!